services/federator/src/Federator/ExternalServer.hs

```diff
@@ -107,14 +107,14 @@ parseRequestData req = do
   when (Wai.requestMethod req /= HTTP.methodPost) $
     throw InvalidRoute
   -- No query parameters are allowed
-  when (not . BS.null . Wai.rawQueryString $ req) $
+  unless (BS.null . Wai.rawQueryString $ req) $
     throw InvalidRoute
   -- check that the path has the expected form
   (componentSeg, rpcPath) <- case Wai.pathInfo req of
     ["federation", comp, rpc] -> pure (comp, rpc)
     _ -> throw InvalidRoute
 
-  when (not (Text.all isAllowedRPCChar rpcPath)) $
+  unless (Text.all isAllowedRPCChar rpcPath) $
```

Comment:
I like this change, but I doubt that it's non-controversial.  :)  Let's see if anybody complains.

Reply:
They seem to be exactly the opposite of each other:

```haskell
when p s = if p then s else pure ()

unless p s =  if p then pure () else s
```

So, I'm wondering how this could be controversial. :thinking:

Reply:
it's clearly still correct, just a question of style: is code easier to read if with fewer keywords, or with less complexity in boolean expressions?  i'm also leaning towards `unless`, and i think we're way too deep into a bike shedding thing again here, so just ignore me.  if somebody feels strongly, they can make a PR with an hlint rule that rolls this back, *then* we can bike shed.  :)